src/System.Net.NameResolution/src/System/Net/NameResolutionPal.Unix.cs

```diff
@@ -72,21 +73,44 @@ private static unsafe IPHostEntry CreateIPHostEntry(Interop.Sys.HostEntry hostEn
             }
             else
             {
-                ipAddresses = new IPAddress[numAddresses];
-
-                void* addressListHandle = hostEntry.AddressListHandle;
-                var nativeIPAddress = default(Interop.Sys.IPAddress);
-                for (int i = 0; i < numAddresses; i++)
+                //
+                // getaddrinfo returns multiple entries per address, for each socket type (datagram, stream, etc.).
+                // Our callers expect just one entry for each address.  So we need to deduplicate the results.
```

Comment:
I'd looked briefly at the calls we're making to Windows, and it appears we're passing similar args on both Unix and Windows.  Is the OS implementation of these functions on Windows already doing such filtering, or is there some fundamental difference in how Windows treats the different socket types?